// @analogjs/vitest-angular/src/lib/builders/build/plugins/angular-memory-plugin.js
"use strict";
/**
 * @license
 * Copyright Google LLC All Rights Reserved.
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://angular.dev/license
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.createAngularMemoryPlugin = createAngularMemoryPlugin;
// import assert from 'node:assert';
const node_path_1 = require("node:path");
async function createAngularMemoryPlugin(options) {
    const { normalizePath } = await Function('return import("vite")')();
    const { outputFiles, external } = options;
    let config;
    let projectRoot;
    const workspaceRoot = options?.workspaceRoot || process.cwd();
    return {
        name: 'vite:angular-memory',
        // Ensures plugin hooks run before built-in Vite hooks
        enforce: 'pre',
        config(userConfig) {
            config = userConfig;
            projectRoot = (0, node_path_1.resolve)(workspaceRoot, config.root || '.');
        },
        async resolveId(source, importer) {
            // Prevent vite from resolving an explicit external dependency (`externalDependencies` option)
            if (external?.includes(source)) {
                // This is still not ideal since Vite will still transform the import specifier to
                // `/@id/${source}` but is currently closer to a raw external than a resolved file path.
                return source;
            }
            if (importer) {
                if (source[0] === '.' &&
                    normalizePath(importer).startsWith(projectRoot)) {
                    // Remove query if present
                    const [importerFile] = importer.split('?', 1);
                    source =
                        '/' + (0, node_path_1.join)((0, node_path_1.dirname)((0, node_path_1.relative)(projectRoot, importerFile)), source);
                }
            }
            const [file] = source.split('?', 1);
            const fileSplits = file.split('/');
            if (outputFiles.has(fileSplits[fileSplits.length - 1])) {
                return fileSplits[fileSplits.length - 1];
            }
            if (outputFiles.has(file)) {
                return (0, node_path_1.join)(projectRoot, source);
            }
            return;
        },
        load(id) {
            const [file] = id.split('?', 1);
            const relativeFile = options.angularVersion < 19
                ? normalizePath((0, node_path_1.relative)(projectRoot, file))
                    .replace(/^.*\//, '')
                    .replace('.ts', '.js')
                : 'spec-' +
                    normalizePath((0, node_path_1.relative)(projectRoot, file))
                        .replace('.ts', '.js')
                        .replace(/^[./]+/, '_')
                        .replace(/\//g, '-');
            const codeContents = outputFiles.get(relativeFile)?.contents ||
                outputFiles.get(id)?.contents;
            if (codeContents === undefined) {
                return undefined;
            }
            const code = Buffer.from(codeContents).toString('utf-8');
            const mapContents = outputFiles.get(relativeFile + '.map')?.contents;
            return {
                // Remove source map URL comments from the code if a sourcemap is present.
                // Vite will inline and add an additional sourcemap URL for the sourcemap.
                code: mapContents
                    ? code.replace(/^\/\/# sourceMappingURL=[^\r\n]*/gm, '')
                    : code,
                map: mapContents && Buffer.from(mapContents).toString('utf-8'),
            };
        },
    };
}
