// luxon-angular/esm2020/lib/formatting/date-time-to-relative-calendar.pipe.mjs
import { Pipe } from '@angular/core';
import { interval } from 'rxjs';
import * as i0 from "@angular/core";
export class DateTimeToRelativeCalendarPipe {
    constructor(ref) {
        this.ref = ref;
    }
    ngOnDestroy() {
        if (this.subscription) {
            this.subscription.unsubscribe();
            this.subscription = null;
        }
    }
    transform(value, opts) {
        if (value == null) {
            if (this.subscription) {
                this.subscription.unsubscribe();
                this.subscription = null;
            }
            return null;
        }
        if (!this.subscription) {
            this.subscription = interval(1000).subscribe(() => this.ref.markForCheck());
        }
        return value.toRelativeCalendar(opts);
    }
}
DateTimeToRelativeCalendarPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.2", ngImport: i0, type: DateTimeToRelativeCalendarPipe, deps: [{ token: i0.ChangeDetectorRef }], target: i0.ɵɵFactoryTarget.Pipe });
DateTimeToRelativeCalendarPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "15.0.2", ngImport: i0, type: DateTimeToRelativeCalendarPipe, name: "dateTimeToRelativeCalendar", pure: false });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.2", ngImport: i0, type: DateTimeToRelativeCalendarPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'dateTimeToRelativeCalendar',
                    pure: false
                }]
        }], ctorParameters: function () { return [{ type: i0.ChangeDetectorRef }]; } });
