// @analogjs/vitest-angular/src/lib/builders/test/vitest.impl.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getExtraArgs = getExtraArgs;
const architect_1 = require("@angular-devkit/architect");
async function vitestBuilder(options, context) {
    process.env['TEST'] = 'true';
    process.env['VITEST'] = 'true';
    const { startVitest } = await Function('return import("vitest/node")')();
    const projectConfig = await context.getProjectMetadata(context.target);
    const { coverageArgs, ...extraArgs } = await getExtraArgs(options);
    const watch = options.watch === true;
    const ui = options.ui === true;
    const coverageEnabled = options.coverage === true;
    const update = options.update === true;
    const config = {
        root: `${projectConfig['root'] || '.'}`,
        watch,
        ui,
        config: options.configFile,
        coverage: {
            enabled: coverageEnabled,
            ...coverageArgs,
        },
        update,
        ...extraArgs,
    };
    const server = await startVitest('test', options.testFiles ?? [], config, {
        test: { watch },
    });
    let hasErrors = false;
    const processExit = () => {
        server?.exit();
        if (hasErrors) {
            process.exit(1);
        }
        else {
            process.exit(0);
        }
    };
    if (options.watch) {
        process.on('SIGINT', processExit);
        process.on('SIGTERM', processExit);
        process.on('exit', processExit);
    }
    // vitest sets the exitCode = 1 when code coverage isn't met
    hasErrors = (process.exitCode && process.exitCode !== 0);
    return {
        success: !hasErrors,
    };
}
async function getExtraArgs(options) {
    // support passing extra args to Vitest CLI
    const schema = await Promise.resolve().then(() => require('./schema.json'));
    const extraArgs = {};
    for (const key of Object.keys(options)) {
        if (!schema.properties[key]) {
            extraArgs[key] = options[key];
        }
    }
    return extraArgs;
}
exports.default = (0, architect_1.createBuilder)(vitestBuilder);
